src/ucp/core/ucp_request.inl

```diff
@@ -475,6 +475,7 @@ ucp_request_recv_data_unpack(ucp_request_t *req, const void *data,
         ucp_request_unpack_contig(req,
                                   UCS_PTR_BYTE_OFFSET(req->recv.buffer, offset),
                                   data, length);
+        req->recv.state.offset += length;
```

Comment:
afaiu, state is not initialized in `ucp_dt_recv_state_init` for contig data.
Plz, check that it does not affect other contig flows in the code. Maybe it is worth to adjust the offset in rkey_ptr progress only?

Reply:
> Maybe it is worth to adjust the offset in rkey_ptr progress only?

good idea, done